frontend/cs/r1cs/builder.go

```diff
@@ -68,6 +72,8 @@ func newBuilder(field *big.Int, config frontend.CompileConfig) *builder {
 		mtBooleans: make(map[uint64][]expr.LinearExpression, config.Capacity/10),
 		config:     config,
 		heap:       make(minHeap, 0, 100),
+		mbuf1:      make(expr.LinearExpression, 0, 100),
```

Comment:
If CompressThreshold is set in config then it would be better to use this as the initial capacity of the buffers?